service/src/test/java/uk/gov/hmcts/reform/fpl/service/orders/validator/SupervisionOrderEndDateValidatorTest.java

```diff
@@ -0,0 +1,210 @@
+                .manageSupervisionOrderEndDateType(SET_CALENDAR_DAY)
+                .manageOrdersSetDateEndDate(dateNow().plusYears(2))
+                .build())
+            .build();
+
+        assertThat(underTest.validate(caseData)).isEqualTo(List.of(TEST_END_DATE_RANGE_MESSAGE));
+    }
+
+    @Test
+    void shouldReturnErrorForDateTimeOverOneYearFromNo() {
+        final LocalDateTime supervisionOrderEndDateTime = time.now();
+        CaseData caseData = CaseData.builder()
+            .manageOrdersEventData(ManageOrdersEventData.builder()
+                .manageSupervisionOrderEndDateType(SET_CALENDAR_DAY_AND_TIME)
+                .manageOrdersSetDateAndTimeEndDate(supervisionOrderEndDateTime.plusMonths(MAXIMUM_MONTHS_ACCEPTED + 1))
+                .build())
+            .build();
+
+        assertThat(underTest.validate(caseData)).isEqualTo(
+            List.of(TEST_END_DATE_RANGE_MESSAGE));
+    }
+
+    @Test
+    void shouldReturnErrorForDateInPast() {
+        CaseData caseData = CaseData.builder()
+            .manageOrdersEventData(ManageOrdersEventData.builder()
+                .manageSupervisionOrderEndDateType(SET_CALENDAR_DAY)
+                .manageOrdersSetDateEndDate(dateNow().minusDays(7))
+                .build())
+            .build();
+
+        assertThat(underTest.validate(caseData)).isEqualTo(List.of(TEST_FUTURE_DATE_MESSAGE));
+    }
+
+    @Test
+    void shouldReturnErrorForDateTimeInPast() {
+        final LocalDateTime supervisionOrderEndDateTime = time.now();
+        CaseData caseData = CaseData.builder()
+            .manageOrdersEventData(ManageOrdersEventData.builder()
+                .manageSupervisionOrderEndDateType(SET_CALENDAR_DAY_AND_TIME)
+                .manageOrdersSetDateAndTimeEndDate(supervisionOrderEndDateTime.minusDays(10))
+                .build())
+            .build();
+
+        assertThat(underTest.validate(caseData)).isEqualTo(List.of(TEST_FUTURE_DATE_MESSAGE));
+    }
+
+    @Test
+    void shouldReturnErrorForNumberOfMonthsOverMaximum() {
+        CaseData caseData = CaseData.builder()
+            .manageOrdersEventData(ManageOrdersEventData.builder()
+                .manageSupervisionOrderEndDateType(SET_NUMBER_OF_MONTHS)
+                .manageOrdersSetMonthsEndDate(MAXIMUM_MONTHS_ACCEPTED + 1)
+                .build())
+            .build();
+
+        assertThat(underTest.validate(caseData)).isEqualTo(List.of(TEST_END_DATE_RANGE_MESSAGE));
+    }
+
+    @Test
+    void shouldReturnErrorForNumberOfMonthsIsLessThanAccepted() {
+        CaseData caseData = CaseData.builder()
+            .manageOrdersEventData(ManageOrdersEventData.builder()
+                .manageSupervisionOrderEndDateType(SET_NUMBER_OF_MONTHS)
+                .manageOrdersSetMonthsEndDate(MINIMUM_MONTHS_ACCEPTED - 1)
+                .build())
+            .build();
+
+        deepEquals(underTest.validate(caseData), TEST_UNDER_DATE_RANGE_MESSAGE);
```

Comment:
is `deepEquals()` working the way you expect here? it didn't pick up the different error message in `SupervisionOrderEndDateValidator` so I wonder if it's giving false posities